### PR TITLE
Bugfixes for lbann sw stack

### PR DESCRIPTION
**Pull request by bvanessen**
Update Aluminum for building without CUDA and DiHydrogen for building with OpenMP on OSX.

### CODE REVIEW

**Review by adamjstewart — COMMENTED** [review]
W.r.t. the macOS Clang OpenMP stuff:

This doesn't make a lot of sense to me. First of all, I think this is specific to `%apple-clang`, I don't think it affects `%clang`. Second, my `clang` is found in `/usr/bin`, but there is no `/usr/lib/libomp.dylib`. I think you want to add a dependency on `llvm-openmp`. See what I did in `py-scikit-learn` for an example.

**Comment by bvanessen** [comment]
> W.r.t. the macOS Clang OpenMP stuff:
> 
> This doesn't make a lot of sense to me. First of all, I think this is specific to `%apple-clang`, I don't think it affects `%clang`. Second, my `clang` is found in `/usr/bin`, but there is no `/usr/lib/libomp.dylib`. I think you want to add a dependency on `llvm-openmp`. See what I did in `py-scikit-learn` for an example.

I will take a look at `py-scikit-learn`, but I am relying on a `brew` installed clang that has `openmp` support rather than allowing `spack` to install clang.

**Comment by adamjstewart** [comment]
That's fine, but I don't think it will work for anyone using Apple Clang.

File: var/spack/repos/builtin/packages/aluminum/package.py [hunk 1]
```diff
@@ -65,8 +65,9 @@ def cmake_args(self):
                 '-DALUMINUM_ENABLE_MPI_CUDA:BOOL=%s' % ('+ht' in spec))
 
         if '@:0.1,0.6.0:':
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/aluminum/package.py]: This is missing an `in spec` or `spec.satisfies()`. Python strings always evaluate to True if non-empty. You can actually combine this line and the one below it.

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — APPROVED** [review]

**Comment by bvanessen** [comment]
@adamjstewart Thanks for all of the feedback on how to make LLVM and OpenMP work better with `apple-clang`  as you can see I am trying to incorporate all of your suggestions to the whole LBANN software stack.

**Pull request merged**